### PR TITLE
Make app id unique across apps started by dapr run

### PR DESCRIPTION
**Pull request by pravinpushkar**
Signed-off-by: Pravin Pushkar <ppushkar@microsoft.com>

# Description

- Make appID unique across apps started by dapr run. Not uniqueness was introduced by mistake in rel-1.6. There is a diff issues opened for supporting this correctly - #698. 
- dapr run -f will exit completely if AppIDs are not unique in `dapr.yaml`

### CODE REVIEW

File: pkg/standalone/run.go [hunk 1]
```diff
@@ -84,6 +84,15 @@ func (meta *DaprMeta) newAppID() string {
 	}
 }
 
+func (config *RunConfig) validateAppID(meta *DaprMeta) error {
+	if config.AppID == "" {
+		config.AppID = meta.newAppID()
+	} else if meta.idExists(config.AppID) {
+		return fmt.Errorf("invalid configuration for AppID. App id %q is not available", config.AppID)
```
Comment (shubham1172) [inline pkg/standalone/run.go:91]:
```suggestion
		return fmt.Errorf("invalid configuration for appID. App ID %q is already in use", config.AppID)
```

**Review by shubham1172 — DISMISSED** [review]
Small nit

**Review by mukundansundar — CHANGES_REQUESTED** [review]
add a test with normal run as well, running an app with app id and the next run `executeWhileRunning` should fail.

Change it to point to `release-1.10`

**Review by mukundansundar — DISMISSED** [review]
lgtm

**Review by shubham1172 — APPROVED** [review]

**Comment by mukundansundar** [comment]
AppID uniqueness was unintentionally removed in 1.6, and there is no complete support for it yet. Will work on https://github.com/dapr/cli/issues/698 to allow appID duplication.

**Review by mukundansundar — APPROVED** [review]

**Review by shubham1172 — DISMISSED** [review]

**Review by mukundansundar — APPROVED** [review]

**Review by shubham1172 — APPROVED** [review]

**Review by artursouza — APPROVED** [review]

**Comment by pravinpushkar** [comment]
one suggested improvement that can be tried for 1.11 until #698 is implemented - 

> You are checking based on the list of processes. It is not guaranteed but it is better than we have now. This can also be achieved by a file lock, but not sure how to do that on Windows. It can be in 1.11 since it would avoid race conditions.

**Comment by mukundansundar** [comment]
@yaron2 had some concerns with this PR breaking some of the users and also that this unintentional change has been there from 1.6. 

@yaron2 thoughts on this?

**Comment by yaron2** [comment]
> @yaron2 had some concerns with this PR breaking some of the users and also that this unintentional change has been there from 1.6.
> 
> @yaron2 thoughts on this?

Technically, it's not forbidden to run two instances of Dapr with the same app Id - this is what happens naturally on K8s with multiple pods all having the same app Id. So unless there is something in the local dev experience that is degraded by having this, maybe it's best to leave it as is, because:

1. It has been allowed in the CLI for the last 3 releases
2. It allows developers to easily test things like two consumers in a consumer group with pub/sub, and multiple consumers of a distributed Lock
3. There may have been users who use the CLI to bootstrap non containerized envs with Dapr who rely on this, although unlikely.

I feel like we can go either way, but would like to revert this based on something concrete.

**Comment by pravinpushkar** [comment]
@yaron2 
> Technically, it's not forbidden to run two instances of Dapr with the same app Id - this is what happens naturally on K8s with multiple pods all having the same app Id. So unless there is something in the local dev experience that is degraded by having this, maybe it's best to leave it as is

If we are going to have this as is then we have to correct the `dapr stop` behaviour.
Currently `dapr stop appID` will stop any one of the many instances of apps having same `appID`.
`dapr stop` iterates through the apps list obtained via `dapr list` and exist after getting the 1st match for the provided app ID. 

And similar behaviour happen for `dapr run -f`.

**Comment by mukundansundar** [comment]
The behavior of the following CLI commands also will be affected right? `dapr stop`, `dapr publish`, `dapr invoke` etc. 
Since all of them use `list` to list the apps and based on the `appID` select a particular app to perform the function of invoke, publish or to stop

**Comment by mukundansundar** [comment]
@dapr/maintainers-cli Based on the discussions that we have had offline, this change will not be done.

**Pull request closed without merging**